cpp/src/arrow/compute/kernels/scalar_cast_test.cc

```diff
@@ -611,6 +949,44 @@ TEST(Cast, FloatingToDecimal) {
   // Edge cases are tested for Decimal128::FromReal()
 }
 
+TEST(Cast, FloatingToDecimal256) {
```

Comment:
Don't you want to avoid the code duplication here? This looks like exactly the same test as above.